src/librustc/mir/visit.rs

```diff
@@ -247,42 +247,30 @@ macro_rules! make_mir_visitor {
 
             fn super_mir(&mut self,
                          mir: & $($mutability)* Mir<'tcx>) {
-                let Mir {
```

Comment:
I don't want people to modify the `basic_blocks` field without invalidating the cache. Maybe make `visit` a submodule of `repr` or something?